diff --git a/ckan/config/middleware.py b/ckan/config/middleware.py
index f0dca73..f5d3a70 100644
--- a/ckan/config/middleware.py
+++ b/ckan/config/middleware.py
@@ -103,7 +103,8 @@ def make_app(global_conf, full_stack=True, static_files=True, **app_conf):
                 who_parser.challenge_decider,
                 logging.getLogger('repoze.who'),
                 logging.WARN, # ignored
-                who_parser.remote_user_key,
+                #who_parser.remote_user_key,
+                'REMOTE_USER_REAL',
            )
     
     # Establish the Registry for this application
@@ -212,7 +213,7 @@ class PageCacheMiddleware(object):
 
         # Only use cache for GET requests
         # REMOTE_USER is used by some tests.
-        if environ['REQUEST_METHOD'] != 'GET' or environ.get('REMOTE_USER'):
+        if environ['REQUEST_METHOD'] != 'GET' or environ.get('REMOTE_USER_REAL'):
             return self.app(environ, start_response)
 
         # If there is a ckan cookie (or auth_tkt) we avoid the cache.
diff --git a/ckan/lib/base.py b/ckan/lib/base.py
index ffb79a7..2d9defa 100644
--- a/ckan/lib/base.py
+++ b/ckan/lib/base.py
@@ -182,7 +182,7 @@ class BaseController(WSGIController):
         # environ['REMOTE_USER'] is set by repoze.who if it authenticates a user's
         # cookie or OpenID. (But it doesn't check the user (still) exists in our
         # database - we need to do that here.
-        c.user = request.environ.get('REMOTE_USER', '')
+        c.user = request.environ.get('REMOTE_USER_REAL', '')
         if c.user:
             c.user = c.user.decode('utf8')
             c.userobj = model.User.by_name(c.user)
