diff --git a/ckan/controllers/datastore.py b/ckan/controllers/datastore.py
index bd5e2d1..d92e766 100644
--- a/ckan/controllers/datastore.py
+++ b/ckan/controllers/datastore.py
@@ -10,6 +10,7 @@ class DatastoreController(BaseController):
         index_name = 'ckan-%s' % g.site_id
         query_string = request.environ['QUERY_STRING']
         redirect = "/elastic/%s/%s%s?%s" % (index_name, id, url, query_string)
+        redirect = "/open-data" + redirect
         # headers must be ascii strings
         response.headers['X-Accel-Redirect'] = str(redirect)
 
